### ISSUE TITLE
Issue with STM32Flash utility

### ISSUE THREAD

**Issue by Mani-Sadhasivam** [RAKWireless/RAK5205-WisTrio-LoRa#3]
Hello,

Flashing the pre built firmware available in the repo using STM32Flash utility fails to start the application. Below is the output of the utility:

```
sudo ./stm32flash -b 115200 -w RAK5205_LF_trackerboard.bin -v -g 0x08000000 -e 255 /dev/ttyUSB0
stm32flash 0.5

http://stm32flash.sourceforge.net/

Using Parser : Raw BINARY
Interface serial_posix: 115200 8E1
Version      : 0x31
Option 1     : 0x00
Option 2     : 0x00
Device ID    : 0x0429 (STM32L1xxx6(8/B)A)
- RAM        : 32KiB  (4096b reserved by bootloader)
- Flash      : 128KiB (size first sector: 16x256)
- Option RAM : 32b
- System RAM : 4KiB
Write to memory
Erasing memory
Wrote and verified address 0x0800ff00 (62.04%) Done.

Starting execution at address 0x08000000... done.
```
Looks like the entire firmware is not getting written into Flash memory (note 64%). However, flashing the Zephyr binary works fine.

**Comment by afremont** [comment]
Are you sure that you want the 433MHz version?  It seems that most people would want the HF version (868MHz or 915MHz) of the firmware, but you might be the exception.

**Comment by Mani-Sadhasivam** [comment]
@afremont I just took that binary as an example. But I can't flash either HF/LF version firmware.

**Comment by afremont** [comment]
Are you using an ST-Link V2 device?  If so, maybe upgrading the firmware in the device will help.  I haven't received my tracker board yet, so I have no experience with flashing it yet.  It should be delivered today.  :)

**Comment by michaelshmitty** [comment]
I ran into the same problem. When I used the [STM32 Flash loader demonstrator](https://www.st.com/en/development-tools/flasher-stm32.html) it worked fine though so it may be an issue with the stm32flash utility.
STM32 Flash loader demonstrator is Windows only though.
I noticed this firmware project is a bit too dependent on a windows environment, which is unfortunate. I do hope RAK will continue to improve and work on a more multiplatform approach in the future.
